test/units/cli/test_galaxy.py

```diff
@@ -630,7 +632,12 @@ def test_invalid_collection_name_init(name):
 ])
 def test_invalid_collection_name_install(name, expected, tmp_path_factory):
     install_path = to_text(tmp_path_factory.mktemp('test-ÅÑŚÌβŁÈ Collections'))
-    expected = "Invalid collection name '%s', name must be in the format <namespace>.<collection>" % expected
+
+    # FIXME: we should add the collection name in the error message
+    # Used to be: expected = "Invalid collection name '%s', name must be in the format <namespace>.<collection>" % expected
+    expected = "Neither the collection requirement entry key 'name', nor 'source' point to a concrete resolvable collection artifact. "
+    expected += "Also 'name' is not an FQCN\. A valid collection name must be in the format <namespace>\.<collection>\. "
+    expected += "Please make sure that the namespace and the collection name  contain characters from \[a\-zA\-Z0\-9_\] only\."
```

Comment:
I think it would helpful in the case of a requirements.yml to include name or source in the error.

Reply:
We can't do this on that level (in the `dataclasses` module). It'd be easier if we fully embraced the layered approach with specific exceptions closer to the core and AnsibleError on the outer layer.

Reply:
Okay, so we'd probably want to parse the requirements the first time closer to core. This exception message is mostly an upgrade, but it makes it harder to find which entry failed.

Reply:
I think that those objects are first created in the cli module and that's where the caller would have to add extra info to this text.